src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -664,6 +665,32 @@ class spell_pri_benediction : public SpellScript
     }
 };
 
+// 368275 - Binding Heals
+class spell_pri_binding_heals : public AuraScript
+{
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return ValidateSpellInfo({ SPELL_PRIEST_BINDING_HEALS_HEAL })
+            && ValidateSpellEffect({ {spellInfo->Id, EFFECT_0} });
+    }
+
+    void HandleEffectProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+    {
+        Unit* caster = eventInfo.GetActor();
+        if (caster == eventInfo.GetProcTarget())
+            return;
+
+        CastSpellExtraArgs args(aurEff);
+        args.AddSpellBP0(CalculatePct(eventInfo.GetHealInfo()->GetHeal(), aurEff->GetAmount()));
+        caster->CastSpell(caster, SPELL_PRIEST_BINDING_HEALS_HEAL, args);
+    }
+
+    void Register() override
+    {
+        OnEffectProc += AuraEffectProcFn(spell_pri_binding_heals::HandleEffectProc, EFFECT_0, SPELL_AURA_DUMMY);
```

Comment:
maybe using DoCheckProc to check GetProcTarget != GetActor instead to prevent going into OnEffectProc unnecessarily